### PR TITLE
gramps: fix map view

### PR DESCRIPTION
**Pull request by panicgh**
###### Motivation for this change

The OSM map view in `gramps` does not work due to  missing dependency.

To reproduce: start `gramps`, do not load a family tree, select the "Geography" panel, try to load new places on the map (it might use tiles from cache). No (new) tiles are loaded and `gramps` outputs a warning message to the terminal:

```
(.gramps-wrapped:1042320): OsmGpsMap-WARNING **: 00:07:59.851: Error downloading tile: 6 - TLS/SSL support not available; install glib-networking
```

Fix in this PR: conditionally add a `glib-networking` dependency if OSM map support is enabled.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
Please change pull request to master. We only put changes causing mass-rebuilds in staging.
Thanks.

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 152395` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gramps</li>
  </ul>
</details>

**Review by davidak — CHANGES_REQUESTED** [review]
Changes look good, issue is fixed

Builds, works

Please rebase to master and change branch in PR!

**Comment by panicgh** [comment]
Is there a way to remove/undo all the automatically requested reviewers due to the PR base change?

**Comment by davidak** [comment]
@panicgh you can request reviews at the top right and also remove them there

this is very annoying, but happen often. not your fault

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 152395` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gramps</li>
  </ul>
</details>

**Comment by panicgh** [comment]
> @panicgh you can request reviews at the top right and also remove them there

I see the reviewers in the top right corner, but I cannot add/remove them. For example, I was only able to re-request review from you, since you did the one before. But I would not have been able to add you from the beginning of the PR, nor could I remove you/unrequest your review now.

**Comment by davidak** [comment]
i see. maybe only committers are allowed to remove them. i did it then

**Review by davidak — APPROVED** [review]
Changes look good, issue is fixed

Builds, works

Screenshot from 2021-12-28 15-08-33

**Comment by bobby285271** [comment]
> Is there a way to remove/undo all the automatically requested reviewers due to the PR base change?

It will always be nice to mark the PR as draft before changing the merge base. GitHub will not request reviews from CODEOWNERS when the PR is marked as draft. `https://github.com/NixOS/nixpkgs/pull/150968` is a good example for this.

**Comment by davidak** [comment]
@bobby285271 nice trick. thanks for sharing

**Comment by jonringer** [comment]
you can also do:

since this change causes over 500 rebuilds, do you mind targeting the staging branch

```
# finding the common merge base will avoid pinging all CODEOWNERs
common=$(git merge-base origin/master origin/staging)
git rebase --onto=$common HEAD~1 # or however many commits you want to pull
git push .. ... --force
```
then change the base branch in the github PR from `master` -> `staging`

See https://nixos.org/nixpkgs/manual/#submitting-changes-staging-branch for more details on staging branch

**Pull request merged**
